testnet/stacks-node/src/event_dispatcher.rs

```diff
@@ -313,70 +318,272 @@ impl RewardSetEventPayload {
+    let conn_opts = ConnectionOptions::default();
+    let http = StacksHttp::new_client(addr, &conn_opts);
+    let mut connection = NetworkConnection::new(http, &conn_opts, None);
+    let mut request_handle = connection
+        .make_request_handle(0, get_epoch_time_secs() + timeout.as_secs(), 0)
+        .map_err(|e| {
+            io::Error::new(
+                io::ErrorKind::Other,
+                format!("Failed to create request handle: {:?}", &e).as_str(),
+            )
+        })?;
+
+    // Step 3: load up the request with the message we're gonna send, and iteratively dump its
+    // bytes from the handle into the socket (the connection does internal buffering and
+    // bookkeeping to deal with the cases where we fail to fill the socket buffer, or we can't send
+    // anymore because the socket buffer is currently full).
+    request
+        .send(&mut request_handle)
+        .map_err(|e| handle_net_error(e, "Failed to serialize request body"))?;
+
+    debug!("send_request(sending data)");
+    loop {
+        let flushed = request_handle
+            .try_flush()
+            .map_err(|e| handle_net_error(e, "Failed to flush request body"))?;
+
+        // send it out
+        let num_sent = connection
+            .send_data(&mut stream)
+            .map_err(|e| handle_net_error(e, "Failed to send socket data"))?;
+
+        debug!(
+            "send_request(sending data): flushed = {}, num_sent = {}",
+            flushed, num_sent
+        );
+        if flushed && num_sent == 0 {
+            break;
+        }
+
+        if Instant::now().saturating_duration_since(start) > timeout {
+            return Err(io::Error::new(
+                io::ErrorKind::WouldBlock,
+                "Timed out while receiving request",
+            ));
+        }
+    }
+
+    // Step 4: pull bytes from the socket back into the handle, and see if the connection decoded
+    // and dispatched any new messages to the request handle.  If so, then extract the message and
+    // check that it's a well-formed HTTP response.
+    debug!("send_request(receiving data)");
+    let response;
+    loop {
+        // get back the reply
+        debug!("send_request(receiving data): try to receive data");
+        match connection.recv_data(&mut stream) {
+            Ok(nr) => {
+                debug!("send_request(receiving data): received {} bytes", nr);
+            }
+            Err(e) => {
+                return Err(handle_net_error(e, "Failed to receive socket data"));
+            }
+        }
+
+        // fullfill the request -- send it to its corresponding handle
+        debug!("send_request(receiving data): drain inbox");
+        connection.drain_inbox();
+
+        // see if we got a message that was fulfilled in our handle
+        debug!("send_request(receiving data): try receive response");
+        let rh = match request_handle.try_recv() {
+            Ok(resp) => {
+                response = resp;
+                break;
```

Comment:
Use `break` to return a value from the loop:

```suggestion
    let response = loop {
        // get back the reply
        debug!("send_request(receiving data): try to receive data");
        match connection.recv_data(&mut stream) {
            Ok(nr) => {
                debug!("send_request(receiving data): received {} bytes", nr);
            }
            Err(e) => {
                return Err(handle_net_error(e, "Failed to receive socket data"));
            }
        }

        // fullfill the request -- send it to its corresponding handle
        debug!("send_request(receiving data): drain inbox");
        connection.drain_inbox();

        // see if we got a message that was fulfilled in our handle
        debug!("send_request(receiving data): try receive response");
        let rh = match request_handle.try_recv() {
            Ok(resp) => {
                break resp;
```

Reply:
Sure